Feat(eos_designs,eos_cli_config_gen): Add role variables to keep temporary files (#6492)
aristanetworks/ansible-avd: docs/porting-guides/6.x.x.md

| `aaa_settings.authorization.dynamic.dot1x_additional_groups` | `dot1x_settings.dynamic_authorization.additional_groups` |
| `aaa_settings.authentication.dot1x.default` | `dot1x_settings.authentication.radius_groups` |
| `avd_6_behaviors` | - |

> | `avd_eos_designs_debug` | `avd_debug` |

Reviewer: Contribution guide metions deprecated variable and should probably be updated?

./docs/contribution/eos_designs_internal_notes.md:56:
    The facts can be inspected in a file per device by running the `arista.avd.eos_designs` role with `-e avd_eos_designs_debug="{{ true }}"`.

Reviewer 2: Updated by Claus

> | `avd_eos_designs_unset_facts` | - |

Reviewer 3: should we also point to avd_debug here, since it would be the only way to the include the files manually?

Author: I would say no.

| `underlay_mutlticast` | `underlay_mutlticast_pim_sm` or `<node_type_key>.defaults.underlay_multicast.pim_sm.enabled` or `<node_type_key>.node_groups[].underlay_multicast.pim_sm.enabled` or `<node_type_key>.node_groups[].nodes[].underlay_multicast.pim_sm.enabled` or `<node_type_key>.nodes[].underlay_multicast.pim_sm.enabled` |
| `underlay_ospf_authentication.message_digest_keys[].key` | `underlay_ospf_authentication.message_digest_keys[].cleartext_key` |

### Removal of `avd_eos_designs_unset_facts`

In previous versions of AVD, `avd_eos_designs_unset_facts` was used to unset `avd_switch_facts` from Ansible facts after the `eos_designs` role completed.

Starting with AVD 6.0.0, this variable has been removed.

The `eos_designs` role now dumps facts to a JSON file in the AVD temporary directory (defined by `avd_tmp_dir`) instead of storing them in Ansible facts. This temporary directory is automatically cleaned up at the end of the role.

To retain access to these facts, set `avd_debug: true`. This prevents the cleanup of the temporary directory, allowing you to inspect facts data.

> If you need to load the facts back into Ansible for use in subsequent playbook tasks, you can do so by adding an explicit task in your playbook:

Reviewer: As mentioned in another comment - whic would only work if this new task is inserted before eos_cli_config_gen task. Maybe we should clarify this..

Reviewer 3: We don't know where they will need those facts. it could be in some other place than before eos_cli_config_gen.

Reviewer 2: I think this is because of the behavior that Alexey described that if you have `avd_debug: true` as an `eos_designs` variable but set `avd_debug: false` (or keep the default) when running `eos_cli_config_gen`, then this will also delete the tmp files for `eos_designs` and so you wont be able to load the facts _after_ you run `eos_cli_config_gen`.

But this would be fixed by decorrelating both.

Reviewer 2: addressed

```yaml
- name: Generate AVD Structured Configurations and Fabric Documentation
  ansible.builtin.import_role:
    name: arista.avd.eos_designs
  vars:
    avd_debug: true

- name: Load AVD facts from temporary directory
  ansible.builtin.include_vars:
    file: "{{ avd_tmp_dir }}/eos_designs/eos_designs_facts.json"
    name: avd_switch_facts
```

### Renaming of `avd_eos_designs_debug` to `avd_debug`

In previous versions of AVD, the `avd_eos_designs_debug` variable was used to dump input variables to files for debugging purposes.

Starting with AVD 6.0.0, this variable has been renamed to `avd_debug`.

```diff
-avd_eos_designs_debug: true
+avd_debug: true
```

The `eos_designs` role now dumps templated inputs, validated inputs, and facts data to JSON files in the AVD temporary directory (defined by `avd_tmp_dir`). This temporary directory is automatically cleaned up at the end of the role.

To retain access to these files, set `avd_debug: true`. This prevents the cleanup of the temporary directory, allowing you to inspect the data used by `eos_designs` for debugging purposes.

!!! note
    `avd_debug` now also works for the `eos_cli_config_gen` role.

!!! note

>     If Ansible Vault is configured, some files in the temporary directory may be encrypted.

Reviewer 2: ```suggestion
    When Ansible Vault is configured, some files in the temporary directory are encrypted.
```

Reviewer 3: instead of "some", maybe we should say
```suggestion
    When Ansible Vault is configured, files holding input variables in the temporary directory will be encrypted.
```

>     To inspect encrypted files, decryption is required. For example:

Reviewer 2: ```suggestion
    `ansible-vault` CLI can be used to inspect the encrypted files. For example:
```

Reviewer 3: I think you are missing a closing `

    ```bash
    ansible-vault view intended/tmp/eos_designs/validated/DC1-LEAF1A.json
    ```

>     Please refer to the [Ansible documentation](https://docs.ansible.com/projects/ansible/latest/vault_guide/index.html) for more details.

Reviewer 2: ```suggestion
    Please refer to the [Ansible Vault documentation](https://docs.ansible.com/projects/ansible/latest/vault_guide/index.html) for more details.
```
